File: null-log/src/null/cleaner.rs
use std::{
    fs::{DirEntry, ReadDir, read_dir},
    io::{Error, StdoutLock, Write, stdout},
    path::PathBuf,
    primitive::{str, u8},
    process::{Command, Output, exit},
    result::{
        Result,
        Result::{Err, Ok},
    },
};

// Clean Logs by Copying the contents of /dev/null to the Log Files
pub fn clean_logs(directory: &str) -> u8 {
    let log_directory: ReadDir = read_dir(directory).unwrap();
    let mut standard_output: StdoutLock = stdout().lock();

    for directory_entry in log_directory {
        let entry: DirEntry = directory_entry.unwrap();
        let path: PathBuf = entry.path();

        if path.is_file() {
            let clean_logs: Result<Output, Error> =
                Command::new("cp").arg("/dev/null").arg(path).output();

            match clean_logs {
                Ok(cleaning) => {
                    writeln!(standard_output, "\x1b[32;1;3mCleaning Log File...\x1b[0m").unwrap();
                    standard_output.write_all(&cleaning.stdout).unwrap();
                    writeln!(standard_output, "{}", cleaning.status).unwrap();
                }
                Err(error) => {
                    eprintln!("\x1b[31;1;3;4mError Cleaning Log Files:\x1b[0m {}", error);
                    exit(1);
                }
            };
        } else {
            writeln!(standard_output, "Skipping...").unwrap();
        }
    }

    return 0;
}

File: null-log/src/null/cleaner.c
#include "./cleaner.h"

#include <dirent.h>
#include <errno.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/stat.h>
#include <unistd.h>

// Check if a Document is a Regular File
int is_file(struct dirent *directory_entry) {
  struct stat status_buffer;
  int file_status = stat(directory_entry->d_name, &status_buffer);
  if (file_status != 0) {
    return S_ISREG(status_buffer.st_mode);
  }
  return file_status;
}

// Clean Logs by Copying the contents of /dev/null to the Log Files
int8_t clean_logs(char log_path[]) {
  DIR *directory = opendir(log_path);
  struct dirent *directory_entry = readdir(directory);
  char *error_message = strerror(errno);
  struct stat status_buffer;

  if (directory == NULL) {
    printf(stderr, "Error Opening Directory: %s\n", error_message);
    printf("Error Value: %d\n", errno);
    printf("\x1b[31;1;3;4mError(1) - Exiting Null Log\x1b[0m");
    exit(1);
  } else {
    while (directory_entry != NULL) {
      if (is_file(directory_entry)) {
        char character_buffer[256] = "";
        printf(character_buffer, "cp /dev/null %s", directory_entry);
        system(character_buffer);
      } else {
        printf(stderr, "Directory Entry is not a File: Skipping...");
      }
    }
  }

  return 0;
}
